core/includes/install.inc

```diff
@@ -725,6 +758,11 @@ function backdrop_rewrite_settings($settings = array()) {
         preg_match('/\$([^ =]*) /', $line, $variable);
         $variable_name = $variable[1];
         if (array_key_exists($variable_name, $settings)) {
+          // If variable_name is 'database', we need to remove the next 5
+          // lines as they are part of the default database array.
+          if ($variable_name == 'database') {
+            $database_line_count = 5;
+          }
```

Comment:
While not ideal, this database-specific rewriting code is fine to me. We can separate out improving this rewriting into a separate issue if `backdrop_rewrite_settings()` needs to be improved. But since it specifically only works with the default values of settings.php, we can accept this assumption that the database variable is 6 lines long.

Reply:
Yes, I couldn't think of a better way to do this, given the way the function works.